app/Services/EDocument/Imports/MindeeEDocument.php

```diff
@@ -0,0 +1,175 @@
+
+        if ($prediction->documentType->value !== 'INVOICE')
+            throw new Exception('Unsupported document type');
+
+        $grandTotalAmount = $prediction->totalAmount->value;
+        $documentno = $prediction->invoiceNumber->value;
+        $documentdate = $prediction->date->value;
+        $invoiceCurrency = $prediction->locale->currency;
+        $country = $prediction->locale->country;
+
+        $expense = Expense::where('amount', $grandTotalAmount)->where("transaction_reference", $documentno)->whereDate("date", $documentdate)->first();
+        if (empty($expense)) {
+            // The document does not exist as an expense
+            // Handle accordingly
+            $expense = ExpenseFactory::create($user->company()->id, $user->id);
+            $expense->date = $documentdate;
+            $expense->user_id = $user->id;
+            $expense->company_id = $user->company->id;
+            $expense->public_notes = $documentno;
+            $expense->currency_id = Currency::whereCode($invoiceCurrency)->first()->id;
+            $expense->save();
+
+            $this->saveDocuments([
+                $this->file,
+                TempFile::UploadedFileFromRaw(strval($result->document), $documentno . "_mindee_orc_result.txt", "text/plain")
+            ], $expense);
+            $expense->saveQuietly();
+
+            $expense->uses_inclusive_taxes = True;
+            $expense->amount = $grandTotalAmount;
+            $counter = 1;
+            foreach ($prediction->taxes as $taxesElem) {
+                $expense->{"tax_amount$counter"} = $taxesElem->amount;
+                $expense->{"tax_rate$counter"} = $taxesElem->rate;
+                $counter++;
+            }
+
+            $vendor = null;
+            $vendor_contact = VendorContact::where("company_id", $user->company()->id)->where("email", $prediction->supplierEmail)->first();
+            if ($vendor_contact)
+                return $vendor = $vendor_contact->vendor;
+
+            if ($vendor)
+                $vendor = Vendor::where("company_id", $user->company()->id)->where("name", $prediction->supplierName)->first();
+
+            if ($vendor) {
+                // Vendor found
+                $expense->vendor_id = $vendor->id;
+            } else {
+                $vendor = VendorFactory::create($user->company()->id, $user->id);
+                $vendor->name = $prediction->supplierName;
+
+                $vendor->currency_id = Currency::whereCode($invoiceCurrency)->first()?->id;
+                $vendor->phone = $prediction->supplierPhoneNumber;
+                // $vendor->address1 = $address_1; // TODO: we only have the full address string from mindee returned
+                // $vendor->address2 = $address_2;
+                // $vendor->city = $city;
+                // $vendor->postal_code = $postcode;
+                $country = Country::where('iso_3166_2', $country)->first()?->id || Country::where('iso_3166_3', $country)->first()?->id || null;
```

Comment:
How about

```php
$country = app('countries')->first(function ($c) use ($country) {
  return $c->iso_3166_2 == $country || $c->iso_3166_3 == $country;
});
```

Reply:
done. both in mindee and in zugferd implementations